templates/component-rust/content/spin.toml

```diff
@@ -0,0 +1,13 @@
+spin_manifest_version = 2
```

Comment:
It feels extremely awkward to create a `spin.toml` for a library component.  Why does a dependency need a Spin manifest file?  It can't be referenced by a trigger.  It can't be referenced by its component ID (although we could allow that of course).  Most of its settings are misleading e.g. setting allowed outbound hosts looks like it grants access but actually does nothing.  Is this just to opt it into getting included `spin build`?

I feel like the ask here is for the manifest to provide for building and referencing library components but I am not sold that mixing them in with "application" components is the way to go right now: perhaps we should step back and ask how we should represent library components in the manifest, then create a template as part of that?